src/core/load-dropin.c

```diff
@@ -63,10 +67,17 @@ static int process_deps(Unit *u, UnitDependency dependency, const char *dir_suff
 
                 /* We don't treat this as an error, especially because we didn't check this for a
                  * long time. Nevertheless, we warn, because such mismatch can be mighty confusing. */
-                r = unit_symlink_name_compatible(entry, basename(target), u->instance);
+                r = path_extract_filename(target, &target_file);
+                if (r < 0) {
+                        log_unit_warning_errno(u, r, "Can't determine file name for dropin target %s, ignoring: %m",
```

Comment:
"determine" sounds slightly spurious for me. How about the following?
`"Failed to extract file name of dropin target %s, ignoring: %m"`